clang-tools-extra/include-cleaner/unittests/FindHeadersTest.cpp

```diff
@@ -587,6 +588,36 @@ TEST_F(HeadersForSymbolTest, AmbiguousStdSymbols) {
   }
 }
 
+TEST_F(HeadersForSymbolTest, AmbiguousStdSymbolsUsingShadow) {
```

Comment:
might be easier to test this in AnalysisTest, rather than writing a new ASTVisitor, but up to you.